### ISSUE TITLE
Exception in QExpansionItem when dynamically importing/code splitting a custom component for header slot

### ISSUE THREAD

**Issue by lukegravitt** [quasarframework/quasar#9547]
**Describe the bug**
Exception in QExpansionItem when using a custom header slot that consists of a single component that is imported dynamically.

**Codepen/jsFiddle/Codesandbox (required)**

```
// my-expansion-item.vue
// import MyComponent from './my-component.vue' // No issue

export default {
  components: {
    /* Use line below with import at top of file */
    /* MyComponent, */

    MyComponent: () => import('./my-component.vue') // Issue present
  }

... Your code here ...

<q-expanstion-item>
  <template v-slot:header>
    <my-component>
  </template>
  <template v-slot:default>
    <p>Hello there</p>
  </template>
</q-expansion-item>
```

**To Reproduce**
1. Load page with above component on it
2. See error

**Expected behavior**
No exception thrown

**Platform (please complete the following information):**
Quasar Version: 1.15.10
@quasar/app Version: 2.2.4
Quasar mode: SPA

  - [x] SPA
  - [ ] SSR
  - [ ] PWA
  - [ ] Electron
  - [ ] Cordova
  - [ ] Capacitor
  - [ ] BEX

Tested on:
  - [x] SPA
  - [ ] SSR
  - [ ] PWA
  - [ ] Electron
  - [ ] Cordova
  - [ ] Capacitor
  - [ ] BEX

OSX
Yarn
Browsers: Chrome

**Comment by MilosPaunovic** [comment]
Hello @lukegravitt,

I've found a solution for your issue and created a PR for it.

**Comment by rstoenescu** [comment]
Improvement will be available in Quasar v1.15.23.
